tests/scripts/thread-cert/Cert_7_1_03_BorderRouterAsLeader.py

```diff
@@ -91,28 +97,85 @@ def test(self):
         self.simulator.go(5)
         self.assertEqual(self.nodes[SED1].get_state(), 'child')
 
-        self.nodes[ED1].start()
+        self.nodes[MED1].start()
         self.simulator.go(5)
-        self.assertEqual(self.nodes[ED1].get_state(), 'child')
+        self.assertEqual(self.nodes[MED1].get_state(), 'child')
+
+        # 2 - N/A
+        # Clear collected messages
+        self.simulator.get_messages_sent_by(LEADER)
+        self.simulator.get_messages_sent_by(MED1)
+        self.simulator.get_messages_sent_by(SED1)
 
         self.nodes[LEADER].add_prefix('2001:2:0:1::/64', 'paros')
         self.nodes[LEADER].add_prefix('2001:2:0:2::/64', 'paro')
         self.nodes[LEADER].register_netdata()
         self.simulator.go(5)
 
+        leader_messages = self.simulator.get_messages_sent_by(LEADER)
+        med1_messages = self.simulator.get_messages_sent_by(MED1)
+        sed1_messages = self.simulator.get_messages_sent_by(SED1)
+
         addrs = self.nodes[SED1].get_addrs()
         self.assertTrue(any('2001:2:0:1' in addr[0:10] for addr in addrs))
         self.assertFalse(any('2001:2:0:2' in addr[0:10] for addr in addrs))
         for addr in addrs:
             if addr[0:10] == '2001:2:0:1' or addr[0:10] == '2001:2:0:2':
                 self.assertTrue(self.nodes[LEADER].ping(addr))
 
-        addrs = self.nodes[ED1].get_addrs()
+        addrs = self.nodes[MED1].get_addrs()
         self.assertTrue(any('2001:2:0:1' in addr[0:10] for addr in addrs))
         self.assertTrue(any('2001:2:0:2' in addr[0:10] for addr in addrs))
         for addr in addrs:
             if addr[0:10] == '2001:2:0:1' or addr[0:10] == '2001:2:0:2':
                 self.assertTrue(self.nodes[LEADER].ping(addr))
 
+        # 3 - Leader
+        msg = leader_messages.next_mle_message(mle.CommandType.DATA_RESPONSE)
+        network_data_tlv = msg.assertMleMessageContainsTlv(mle.NetworkData)
+        prefixes = [tlv for tlv in network_data_tlv.tlvs if isinstance(tlv, Prefix)]
+        self.assertTrue(len(prefixes) >= 2)
+        for prefix in prefixes:
+            self.assertTrue(contains_tlv(prefix.sub_tlvs, BorderRouter))
+            self.assertTrue(contains_tlv(prefix.sub_tlvs, LowpanId))
+
+        # 4 - N/A
+        msg = med1_messages.next_mle_message(mle.CommandType.CHILD_UPDATE_REQUEST)
```

Comment:
reuse `check_child_update_request_by_child()` in #3387 ?

Reply:
Test Plan requires nothing todo for step 4. Code here is to get addresses registered by MED, and later verify if leader includes those addresses in response.

It would be strange to require `check_child_update_request_by_child ` returns an address list,
another `check_addresses_registration` could be better.